### PR TITLE
DSND-3116: Update pom dependencies

### PR DESCRIPTION
**Pull request by kinpang-CH**
* Seperate sections into internal and external
* Commands `make build test-unit` and `make build test-integration` both builds successfully.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -15,34 +15,34 @@
 
 	<properties>
 		<java.version>21</java.version>
-
 		<main.class>uk.gov.companieshouse.charges.data.ChargesDataApiApplication</main.class>
 
+		<!-- internal dependencies -->
+		<spring.boot.version>3.3.5</spring.boot.version>
```
Comment (sthompsonCH) [inline pom.xml:21]: Can this be 3.4.0?

Comment (sthompsonCH) [reply pom.xml:21]: And the parent pom version can be 2.1.10 I think

Comment (kinpang-CH) [reply pom.xml]: Resolved in latest commit

**Review by cgregoryCH — DISMISSED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Review by cgregoryCH — DISMISSED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cgregoryCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by crowlandsCH — APPROVED** [review]

**Pull request merged**